### ISSUE TITLE
[Bug]: Edge cases in models

### ISSUE THREAD

**Issue by ashwinvaidya17** [openvinotoolkit/anomalib#1478]
### Describe the bug

This issue is to track edge cases in the models.

## Rkde, Dfkde

This basically affects all the models that rely on pca. If the dataset size is less than the `n_component` size then the models fail

The solution should be to throw a message about minimum dataset requirements.

## Efficient AD

The learning rate scheduler is sensitive to the combination of parameters. When the `max_epoch` is 1 and the batch size is greater than the number of data points in the training dataset (`len(training_dataset)`) the step size is returned as 0. Also, we need to provide `max_steps` parameter.

## Task List

-  [ ] Add tests to check these combinations after the fix

### Dataset

MVTec

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

See description

### OS information

OS information:
- OS: [e.g. Ubuntu 20.04]
- Python version: [e.g. 3.8.10]
- Anomalib version: on `feature/custom_trainer` branch
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: [e.g. 2x GeForce RTX 3090]
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

All models should work

### Pip/GitHub

GitHub

### What version/branch did you use?

feature/custom_trainer

### Configuration YAML

```yaml
NA
```

### Logs

```shell
NA
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
